src/main/java/org/ibp/api/rest/labelprinting/GermplasmLabelPrinting.java

```diff
@@ -0,0 +1,515 @@
+import org.ibp.api.rest.labelprinting.domain.LabelsData;
+import org.ibp.api.rest.labelprinting.domain.LabelsGeneratorInput;
+import org.ibp.api.rest.labelprinting.domain.LabelsInfoInput;
+import org.ibp.api.rest.labelprinting.domain.LabelsNeededSummary;
+import org.ibp.api.rest.labelprinting.domain.LabelsNeededSummaryResponse;
+import org.ibp.api.rest.labelprinting.domain.OriginResourceMetadata;
+import org.ibp.api.rest.labelprinting.domain.Sortable;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.support.ResourceBundleMessageSource;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Component
+@Transactional
+public class GermplasmLabelPrinting extends LabelPrintingStrategy {
+
+	public final static String GERMPLASM_DATE = "GERMPLASM DATE";
+	public final static String METHOD_ABBREV = "METHOD ABBREV";
+	public final static String METHOD_NUMBER = "METHOD NUMBER";
+	public final static String METHOD_GROUP = "METHOD GROUP";
+	public final static String PREFERRED_NAME = "PREFERRED NAME";
+	public final static String PREFERRED_ID = "PREFERRED ID";
+	public final static String GROUP_SOURCE_GID = "GROUP SOURCE GID";
+	public final static String GROUP_SOURCE = "GROUP SOURCE";
+	public final static String IMMEDIATE_SOURCE_GID = "IMMEDIATE SOURCE GID";
+	public final static String IMMEDIATE_SOURCE = "IMMEDIATE SOURCE";
+
+	public final static String GID = "GID";
+	public final static String GROUP_ID = "GROUP ID";
+
+
+	@Autowired
+	private ResourceBundleMessageSource messageSource;
+
+	@Autowired
+	private GermplasmDataManager germplasmDataManager;
+
+	@Autowired
+	private SearchRequestService searchRequestService;
+
+	@Autowired
+	private GermplasmSearchService germplasmSearchService;
+
+	@Autowired
+	private GermplasmService germplasmService;
+
+	public static List<FileType> SUPPORTED_FILE_TYPES = Arrays.asList(FileType.CSV, FileType.PDF, FileType.XLS);
+
+	public static List<Sortable> SORTED_BY = Arrays.asList(new Sortable(GERMPLASM_DETAILS_FIELD.GID.name(), GermplasmLabelPrinting.GID),
+		new Sortable(GERMPLASM_DETAILS_FIELD.PREFERRED_NAME.name(), GermplasmLabelPrinting.PREFERRED_NAME),
+		new Sortable(GERMPLASM_DETAILS_FIELD.GROUP_ID.name(), GermplasmLabelPrinting.GROUP_ID),
+		new Sortable(GERMPLASM_DETAILS_FIELD.CREATION_DATE.name(), GermplasmLabelPrinting.GERMPLASM_DATE));
+
+
+	private enum GERMPLASM_DETAILS_FIELD {
+		GID(TermId.GID.getId(), "GID"),
```

Comment:
should we move all the strings to a property file?

Reply:
Maybe. We Actually have the LabelPrintingStaticField to store statics ids for label printing. I'm used when the label doesn't exists as TermId and they represent something generics. e.g. Location.

Reply:
I got lost with this comment. I meant to move the hardcoded strings to messages_en, as we already have in ObservationDatasetLabelPrinting